service/controller/v23/resource/workerasgname/resource.go

```diff
@@ -95,7 +132,23 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster worker ASG name in the cloud formation stack")
 	}
 
-	controllerCtx.Status.Drainer.WorkerASGName = workerASGName
+	// Store the ASG name in the CR status.
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updating CR status")
+
+		customObject.Status.AWS.AutoScalingGroup.Name = workerASGName
+
+		_, err = r.g8sClient.ProviderV1alpha1().AWSConfigs(customObject.Namespace).UpdateStatus(&customObject)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updated CR status")
+
+		// TODO: test if this is needed.
```

Comment:
It is when the object is changed.

Reply:
Good catch, I tested it and forgot to remove the TODO.